taichi/runtime/llvm/runtime.cpp

```diff
@@ -525,8 +525,11 @@ struct LLVMRuntime {
   host_printf_type host_printf;
   host_vsnprintf_type host_vsnprintf;
   Ptr program;
-  Ptr root;
-  size_t root_mem_size;
+  
+  Ptr roots[taichi_max_num_snodes];
```

Comment:
nit: Let's make this `kTaichiMaxNumRoots`? (We can also reduce that number a bit..)